src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/OutgoingTransfersControllerTests.cs

```diff
@@ -476,5 +669,204 @@ public async Task SubmittedTransferPost_ShouldReturnView()
+
+            var categories = transferEvidenceNoteData.CategoryIds;
+
+            //act
+            await outgoingTransferEvidenceController.EditTransferFrom(organisationId, TestFixture.Create<Guid>());
+
+            //assert
+            A.CallTo(() => weeeClient.SendAsync(A<string>._,
+                    A<GetEvidenceNotesForTransferRequest>.That.Matches(r =>
+                        r.OrganisationId == organisationId &&
+                        r.Categories.SequenceEqual(categories))))
+                .MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task EditTransferFromGet_GivenTransferNoteAndCategories_ModelMapperShouldBeCalled()
+        {
+            //arrange
+            A.CallTo(() => weeeClient.SendAsync(A<string>._,
+                A<GetTransferEvidenceNoteForSchemeRequest>._)).Returns(transferEvidenceNoteData);
+
+            A.CallTo(() => weeeClient.SendAsync(A<string>._,
+                A<GetEvidenceNotesForTransferRequest>._)).Returns(evidenceNoteData);
+
+            //act
+            await outgoingTransferEvidenceController.EditTransferFrom(organisationId, TestFixture.Create<Guid>());
+
+            //assert
+            A.CallTo(() => mapper.Map<TransferEvidenceNotesViewModelMapTransfer, TransferEvidenceNotesViewModel>(
+                    A<TransferEvidenceNotesViewModelMapTransfer>.That.Matches(t => t.TransferAllTonnage == false &&
+                        t.TransferEvidenceNoteData ==
+                        transferEvidenceNoteData &&
+                        t.Request == null &&
+                        t.Notes.SequenceEqual(evidenceNoteData) &&
+                        t.OrganisationId == organisationId)))
+                .MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task EditTransferFromGet_GivenMappedModel_ModelShouldBeReturned()
+        {
+            //arrange
+            var model = TestFixture.Create<TransferEvidenceNotesViewModel>();
+            A.CallTo(() => mapper.Map<TransferEvidenceNotesViewModelMapTransfer, TransferEvidenceNotesViewModel>(
+                A<TransferEvidenceNotesViewModelMapTransfer>._)).Returns(model);
+
+            //act
+            var result =
+                await outgoingTransferEvidenceController.EditTransferFrom(organisationId, TestFixture.Create<Guid>()) as
+                    ViewResult;
+
+            //assert
+            result.Model.Should().Be(model);
+        }
+
+        [Fact]
+        public async Task EditTransferFromGet_ShouldReturnView()
+        {
+            //act
+            var result =
+                await outgoingTransferEvidenceController.EditTransferFrom(organisationId, TestFixture.Create<Guid>()) as
+                    ViewResult;
+
+            //assert
+            result.ViewName.Should().Be("EditTransferFrom");
+        }
+
+        [Fact]
+        public async Task EditTransferFromPost_GivenInvalidModel_ViewShouldBeReturned()
```

Comment:
got a test to check for the Post Attribute?

Reply:
This one  EditTransferFrom_ShouldHaveHttpPostAttribute

Reply:
alright, missed it